src/Razor/src/Microsoft.VisualStudio.Editor.Razor/DefaultDynamicDocumentContainer.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis.Razor.ProjectSystem;
+using Microsoft.CodeAnalysis.Razor.Workspaces;
+
+namespace Microsoft.CodeAnalysis.Razor
+{
+    // This types purpose is to serve as a non-Razor specific document delivery mechanism for Roslyn.
+    internal sealed class DefaultDynamicDocumentContainer : DynamicDocumentContainer
+    {
+        private readonly DocumentSnapshot _documentSnapshot;
+        private RazorDocumentExcerptService _excerptService;
+        private RazorSpanMappingService _mappingService;
+
+        public DefaultDynamicDocumentContainer(DocumentSnapshot documentSnapshot)
+        {
+            if (documentSnapshot is null)
+            {
+                throw new ArgumentNullException(nameof(documentSnapshot));
+            }
+
+            _documentSnapshot = documentSnapshot;
+        }
+
+        public override string FilePath => _documentSnapshot.FilePath;
```

Comment:
If this is a bucket of documents what does this property do? Might be useful to leave a doccomment to that effect.

Reply:
I'll leave a top-level comment on this class to indicate its purpose